.github/workflows/integration-test.yaml

```diff
@@ -0,0 +1,66 @@
+name: IntegrationTests
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  pull_request:
+    paths-ignore:
+    - 'Documentation/**'
+    - 'test/**'
+  push:
+    branches:
+    - master
+    - ft/master/**
+    paths-ignore:
+    - 'Documentation/**'
+    - 'test/**'
+
+permissions: read-all
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.after }}
+  cancel-in-progress: true
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 120
+    steps:
+    - name: Checkout master branch to access local actions
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      with:
+        ref: ${{ github.event.repository.default_branch }}
+        persist-credentials: false
+    - name: Set Environment Variables
+      uses: ./.github/actions/set-env-variables
+
+    - name: Set image tag
+      id: vars
+      run: |
+        if [ ${{ github.event.pull_request }} ]; then
+          SHA=${{ github.event.pull_request.head.sha }}
+        else
+          SHA=${{ github.sha }}
+        fi
+        echo ::set-output name=sha::${SHA}
+
+    - name: Checkout
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      with:
+        ref: ${{ steps.vars.outputs.sha }}
+        persist-credentials: false
+
+    - name: Install Go
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568
+      with:
+        go-version: 1.19.4
+
+    - name: Prepare environment
+      timeout-minutes: 30
```

Comment:
nit, do we need 30m for this timeout? Should be doable in 5-10m.

Reply:
good catch :100: